Terminal.Gui/View/View.cs

```diff
@@ -106,15 +106,8 @@ namespace Terminal.Gui;
 
 #endregion API Docs
 
-public partial class View : Responder, ISupportInitializeNotification
+public partial class View : Responder, ISupportInitializeNotification, INotifyPropertyChanging, INotifyPropertyChanged
```

Comment:
I've written this several times regarding `INotifyPropertyChanging`. Not sure if you didn't see it or don't believe me or disagree. 

`INotifyPropertyChanging` does not do what many people think it does. It does NOT prevent the change from happening, only allows you to record that it is happening.

https://github.com/gui-cs/Terminal.Gui/issues/3561#issuecomment-2194932007

https://github.com/gui-cs/Terminal.Gui/pull/3625#issuecomment-2265847460

I don't think we should be using it.

Reply:
I think there might be a little bit of a misinterpretation of what the INotify* interfaces and events are for and do.

They're by no means a replacement for dedicated change events for specific properties, for like 100% of the reasons you mention.

As the name implies, it is a notification. It is informational _only_ and is there to be used by interested consumers regardless of what else may happen. That's also why it is intentionally not cancelable. If you need to know something happened, allowing some other unknown piece of code to hide it from you is black box behavior that is expected in cancelable change events, but very much not desirable for an observer for myriad purposes. Even simple logging uses or things like telemetry are common reasons for a consumer wanting to use IN* events.

And remember - we are not the application. We are used by an application.

Both of the INPC events and the INCC event are very useful in a lot of cases. When something needs to know _that_ something is going to change or has changed, that's what it's there for. And that's why, yes, it's used by WPF/WinUI for data binding - because it's the simplest way to achieve what that feature needs without making any further assumptions and without allowing other code to interfere with it.

Do I think we ourselves should be _consuming_ those events? No, not in almost any case I can think of.

But, should we provide it for consumers, especially since it's a trivial thing to implement and carries no baggage for us? Absolutely.

Dedicated change events are a different beast entirely and are important to have for specific functionality, and I'm glad we have the ones we do or I'd push to add them as well. In fact, if/when I eventually get to the event work I want to do, there's a decent chance there may be a few more such events I'll want to add for sake of completeness.

Not providing such a simple yet useful feature for consumers does them a disservice, if you ask me. And it costs us next to nil. 🤷‍♂️

Hopefully that makes my intentions and the purpose of all that clearer? If not, or if you have any other concerns, I'm more than happy to continue talking about it. 🙂

Reply:
As for the whole XAML thing I've brought up before:

That's actually a perfect example of a reason TO have it - especially since events and interfaces are the kind of thing can't be bolted on by a consumer without modifying the TG source itself.

I don't think XAML functionality belongs in the core Terminal.Gui library at all for many reasons, not the least of which is it would be quite a bit of bloat for anyone not wanting/needing to use that feature. My intention for that was to write a separate library myself that uses TG as the console adapter/engine/framework that renders the markup. It's almost definitely going to end up quite similar to how WPF works, where it's markup that results in source-generated C# that is the actual compiled code.

Reply:
Performance-wise...

If it were some heavy thing that had a negative impact on the function or performance of the core library? Then yeah definitely I wouldn't be in favor of it.

But it really isn't.

In fact, since it operates via the core implementation of .net and uses compile-time constants, it's also pretty much the lightest-weight way to notify anything of anything. Those string names are, in .net terms, already interned at compile time, which turns into a blob in the data section of the binary at JIT time, able to be referenced as `const wchar_t* const` basically.

If a consumer takes action based on the string value, they're still comparing against those interned strings, and most likely themselves also using an implicit constant string, so even in that case it's still extremely lightweight. It's a method call passing one pointer to static data in the compiled binary.

Reply:
Another key point about what it does, which isn't served well by other events:

It allows an observer to know that _anything_ in the public API surface of a type implementing it has changed without needing to subscribe to every individual event, and without us needing to publish events for every single property.

There are plenty of properties across the library that don't have explicit events tied to them. How does an interested observer find out if those changed? They can't, because they have no hook into the implementation to do so.

Any means of doing it without there already being a mechanism provided by TG would require either modifying TG code or using an interceptor or other form of reflection-based manipulation of the types, which isn't fun or pretty.

Reply:
There's also `IObserver<T>` and `IObservable<T>`, which came along later but do the same sort of thing via subscription methods and callback delegates, which is the same idea and often actually implemented on top of events under the hood to avoid duplication.

Now, as for what a consumer can do with the events? Anything they want. And we don't need to care nor should we care or interfere. We're just letting them know, if they care.

And if they want access to the sender of the event, for whatever purpose that we also have no business caring about, it's right there in the delegate signature for the event (`object? sender`), and they are free to use or abuse things via that as they please, too.

And there's one last thing that comes to mind for now that I'll add after this comment just to keep things in chunks...

Reply:
OK, so last thing for now:

***Especially*** since we have no hooks/facilities for logging in TG at present, something like INPC is kind of a bare minimum thing we can do to allow a consumer to log activity from the library and their use of it, without us adding any logging facilities (though I _really_ think we should do that anyway at some point - maybe in a feature release?).

That at least gives a consumer a way of being able to log when things happen, whether caused by their explicit property accesses or by internal TG code accessing those properties. Without it, the consumer pretty much has a black box for anything not publishing a dedicated event.

Lack of logging/visibility outside of a debugger was actually a big enough gripe for me in various usages I have of TG v1 that I have an internal modified version of TG that just has a static ILogger (most often backed by NLog) bolted onto every class, so I can quickly add logging for things I need, either in debugging OR a live running instance of an application.